### PR TITLE
Feat 6061 delete all user sessions on blocking a user

### PR DESCRIPTION
**Pull request by SophieNguyen113**
## What does this PR do?

This PR introduces updates to the Update user status API to include the functionality of deleting all sessions when a user is blocked. Previously, user sessions were not deleted, which caused issues when a different user attempted to sign in on the same device. The changes ensure that when a user's status is updated to blocked, all their active sessions are deleted, allowing a different user to sign in.

Fixes: https://github.com/appwrite/appwrite/issues/6061

## Team
- [Sophie Nguyen](https://github.com/SophieNguyen113)
- [Nha Pham](https://github.com/nhapham03)
- [Rojee Koju](https://github.com/rojeekoju)
- [Edale Miguel](https://github.com/EJM143)

## Test Plan

1. **Verify User Existence**
   - Send a GET request to `/users/{userId}` and check for a 200 status code.

2. **Activate User**
   - PATCH `/users/{userId}/status` with `status: true`.
   - Verify response: status code 200 and body contains `status: true`.

3. **Block User**
   - PATCH `/users/{userId}/status` with `status: false`.
   - Verify response: status code 500.

4. **Check Block Status**
   - GET `/users/{userId}`.
   - Verify response: status code 200 and body contains `status: false`.

5. **Reactivate User**
   - PATCH `/users/{userId}/status` with `status: true`.
   - Verify response: status code 200 and body contains `status: true`.

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Comment by SophieNguyen113** [comment]
@stnguyen90  Hello, since I accidentally messed up my old PR, I would close it and open this new PR. This new one should be the most updated. Please review. Thanks.

**Comment by SophieNguyen113** [comment]
@stnguyen90  Fixed error. Please review. Thanks.

File: tests/e2e/Services/Account/AccountCustomClientTest.php [hunk 1]
```diff
@@ -1314,7 +1315,17 @@ public function testBlockedAccount(): array
             'status' => false,
         ]);
 
+        $this->assertEquals(500, $response['headers']['status-code']);
```
Comment (stnguyen90) [inline tests/e2e/Services/Account/AccountCustomClientTest.php:1318]: Why does this return 500?

File: tests/e2e/Services/Users/UsersBase.php [hunk 2]
```diff
@@ -1038,29 +1038,65 @@ public function testUpdateUserPassword(array $data): array
...
      */
-    #[Retry(count: 1)]
     public function testUpdateUserStatus(array $data): array
     {
-        /**
-         * Test for SUCCESS
-         */
-        $user = $this->client->call(Client::METHOD_PATCH, '/users/' . $data['userId'] . '/status', array_merge([
+        // Verify that the user exists
+        $response = $this->client->call(Client::METHOD_GET, '/users/' . $data['userId'], array_merge([
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:1044]: Why do we need to test the user exists? This depends on testGetUser, so the user should already exist.

File: tests/e2e/Services/Users/UsersBase.php [hunk 3]
```diff
@@ -1038,29 +1038,65 @@ public function testUpdateUserPassword(array $data): array
...
+        $response = $this->client->call(Client::METHOD_PATCH, '/users/' . $data['userId'] . '/status', array_merge([
             'content-type' => 'application/json',
             'x-appwrite-project' => $this->getProject()['$id'],
         ], $this->getHeaders()), [
             'status' => false,
         ]);
 
-        $this->assertEquals($user['headers']['status-code'], 200);
-        $this->assertEquals($user['body']['status'], false);
+        $this->assertEquals($response['headers']['status-code'], 500);
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:1079]: Why does this return 500?

File: tests/e2e/Services/Users/UsersBase.php [hunk 4]
```diff
@@ -1038,29 +1038,65 @@ public function testUpdateUserPassword(array $data): array
...
+        ], $this->getHeaders()));
+
+        $this->assertEquals($response['headers']['status-code'], 200);
+
+        // Test: Activate user
+        $response = $this->client->call(Client::METHOD_PATCH, '/users/' . $data['userId'] . '/status', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+        ], $this->getHeaders()), [
+            'status' => true
```
Comment (stnguyen90) [inline tests/e2e/Services/Users/UsersBase.php:1056]: No need to do this because they are already active and the end of the test tests for activating.

File: tests/e2e/Services/Account/AccountCustomClientTest.php [hunk 5]
```diff
@@ -1306,6 +1306,7 @@ public function testBlockedAccount(): array
 
         $this->assertEquals(200, $response['headers']['status-code']);
 
+        // Block user if status is false
```
Comment (stnguyen90) [inline tests/e2e/Services/Account/AccountCustomClientTest.php:1309]:
```suggestion
        // Block user
```

File: docs/references/users/update-user-status.md [hunk 6]
```diff
@@ -1 +1 @@
-Update the user status by its unique ID. Use this endpoint as an alternative to deleting a user if you want to keep user's ID reserved.
\ No newline at end of file
+Update the user status by its unique ID. This endpoint deletes all active sessions when a user is blocked, resolving issues when a different user attempts to sign in on the same device.
```
Comment (stnguyen90) [inline docs/references/users/update-user-status.md:1]: Looks like part of the old text was removed. Would you please put that back? 

Also, no need for the part about resolving.

```suggestion
Update the user status by its unique ID. Use this endpoint as an alternative to deleting a user if you want to keep user's ID reserved.

When a user is blocked, all their sessions will also be deleted.
```

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: tests/e2e/Services/Account/AccountCustomClientTest.php (hunk 1 above)
Comment (SophieNguyen113) [reply tests/e2e/Services/Account/AccountCustomClientTest.php:1318]: Shouldn't it be 500 because the user is blocked?

File: tests/e2e/Services/Users/UsersBase.php (hunk 3 above)
Comment (SophieNguyen113) [reply tests/e2e/Services/Users/UsersBase.php:1079]: Shouldn't it be 500 because the user is blocked?

File: tests/e2e/Services/Account/AccountCustomClientTest.php (hunk 1 above)
Comment (stnguyen90) [reply tests/e2e/Services/Account/AccountCustomClientTest.php:1318]: No, 500 is a server error. It should return 200 for success

File: tests/e2e/Services/Users/UsersBase.php (hunk 3 above)
Comment (stnguyen90) [reply tests/e2e/Services/Users/UsersBase.php:1079]: No, 500 is a server error. It should return 200 for success

File: tests/e2e/Services/Users/UsersBase.php (hunk 4 above)
Comment (stnguyen90) [reply tests/e2e/Services/Users/UsersBase.php:1056]: This hasn't been addressed

File: tests/e2e/Services/Users/UsersBase.php (hunk 2 above)
Comment (stnguyen90) [reply tests/e2e/Services/Users/UsersBase.php:1044]: This hasn't been addressed

File: tests/e2e/Services/Webhooks/WebhooksCustomServerTest.php
Comment (stnguyen90) [inline tests/e2e/Services/Webhooks/WebhooksCustomServerTest.php:1]: Please revert the changes in this file. The tests are there to verify the webhook sends the blocked user as the payload.

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

**Comment by stnguyen90** [comment]
@SophieNguyen113, are you still working on this issue? If not or there's no activity from you, I'll have to close this PR and reassign the issue.

**Comment by stnguyen90** [comment]
Closing stale PR
